src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -1657,9 +1662,17 @@ private CallResult ExecuteCall<TTracingInstructions>(EvmState vmState, ReadOnlyM
                         if (!stack.PopUInt256(out a)) goto StackUnderflow;
                         long number = a > long.MaxValue ? long.MaxValue : (long)a;
 
-                        Hash256 blockHash = spec.IsEip2935Enabled
-                            ? GetBlockHashFromState(number, blkCtx.Header.Number)
-                            : _blockhashProvider.GetBlockhash(blkCtx.Header, number);
+                        Hash256? blockHash;
+                        try
+                        {
+                            blockHash = spec.IsEip2935Enabled
+                                ? GetBlockHashFromState(number, blkCtx.Header.Number)
+                                : _blockhashProvider.GetBlockhash(blkCtx.Header, number);
+                        }
+                        catch (OutOfGasException)
+                        {
+                            goto OutOfGas;
+                        }
```

Comment:
i dont understand, why is this inside a try-catch, i dont think there should be any outOfGas exception here